File: src/main/java/pl/mroz/buddiesapi/domain/rental/Rental.java
package pl.mroz.buddiesapi.domain.rental;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NonNull;
import lombok.ToString;
import pl.mroz.buddiesapi.domain.common.Location;

import java.time.Instant;
import java.util.UUID;

@Getter
@EqualsAndHashCode
@ToString
@AllArgsConstructor
@Builder(access = AccessLevel.PRIVATE)
public class Rental {

    @NonNull
    private UUID rentalId;

    @NonNull
    private final String title;

    private boolean isNegotiable;

    @NonNull
    private String description;

    @NonNull
    private Location location;

    private int price;

    private int deposit;

    private int rooms;

    private int floor;

    private double size;

//    private double pricePerM;

    private int buildYear;

    @NonNull
    private Instant rentDate;

//    @NonNull
//    private Set<String> featureTags;
//
//    @NonNull
//    private Set<String> photoUrls;

    public static Rental fromDb(RentalRepository.IRentalEntity entity) {
        return Rental.builder()
                .rentalId(entity.getRentalId())
                .title(entity.getTitle())
                .isNegotiable(entity.isNegotiable())
                .description(entity.getDescription())
                .location(Location.fromDb(entity.getLocationEntity()))
                .price(entity.getPrice())
                .deposit(entity.getDeposit())
                .rooms(entity.getRooms())
                .floor(entity.getFloor())
                .size(entity.getSize())
//                .pricePerM(entity.getPricePerM())
                .buildYear(entity.getBuildYear())
                .rentDate(entity.getRentDate())
//                .featureTags(entity.getFeatureTags())
//                .photoUrls(entity.getPhotoUrls())
                .build();
    }

}

File: src/main/kotlin/buddiesapi/domain/rental/Rental.kt
package buddiesapi.domain.rental

import buddiesapi.domain.account.Account
import buddiesapi.domain.common.Location
import java.time.Instant
import java.util.UUID


data class Rental(
    val title: String,
    val rentalId: UUID,
    val author: Account,
    val isNegotiable: Boolean,
    val description: String,
    val location: Location,
    val price: Int,
    val deposit: Int,
    val rooms: Int,
    val floor: Int,
    val size: Double,
    val buildYear: Int,
    val rentDate: Instant,
//    val featureTags: List<String>,
//    val photoUrls: List<String>,
) {

    companion object {
        fun fromDb(entity: RentalRepository.IRentalEntity): Rental {
            return entity.let {
                Rental(
                    it.title,
                    it.rentalId!!,
                    Account.fromDb(it.authorEntity),
                    it.isNegotiable,
                    it.description,
                    Location.fromDb(it.locationEntity),
                    it.price,
                    it.deposit,
                    it.rooms,
                    it.floor,
                    it.size,
                    it.buildYear,
                    it.rentDate
                )
            }
        }
    }
}
